src/testing/support/assertRender.ts

```diff
@@ -98,9 +99,67 @@ function formatNode(node: WNode | VNode, tabs: any): string {
 	return `v("${node.tag}", ${properties}`;
 }
 
+const assertionWidgets = [
+	{
+		type: Ignore,
+		value(actual: DNode, expected: DNode) {
+			const node = actual ? actual : expected;
+			return [actual, node];
+		}
+	}
+];
+
+function isNode(node: any): node is VNode | WNode {
+	return isVNode(node) || isWNode(node);
+}
+
+function decorate(actual: DNode | DNode[], expected: DNode | DNode[]): [DNode[], DNode[]] {
+	actual = Array.isArray(actual) ? actual : [actual];
+	expected = Array.isArray(expected) ? expected : [expected];
+	let actualDecoratedNodes = [];
+	let expectedDecoratedNodes = [];
+	const length = actual.length > expected.length ? actual.length : expected.length;
+	for (let i = 0; i < length; i++) {
+		let actualNode = actual[i];
+		let expectedNode = expected[i];
+
+		if (isWNode(expectedNode)) {
+			[actualNode, expectedNode] = assertionWidgets.reduce(
+				(result, assertionWidget) => {
+					if ((expectedNode as any).widgetConstructor === assertionWidget.type) {
+						return assertionWidget.value(result[0], result[1]);
+					}
+					return [result[0], result[1]];
+				},
+				[actualNode, expectedNode]
+			);
+		}
+		if (isNode(expectedNode)) {
+			if (typeof expectedNode.properties === 'function') {
+				const actualProperties = isNode(actualNode) ? actualNode.properties : {};
+				expectedNode.properties = expectedNode.properties(actualProperties);
+			}
+		}
+		const childrenA = isNode(actualNode) ? actualNode.children : [];
+		const childrenB = isNode(expectedNode) ? expectedNode.children : [];
+
+		const [actualChildren, expectedChildren] = decorate(childrenA, childrenB);
+		if (isNode(actualNode)) {
+			actualNode.children = actualChildren;
+		}
+		if (isNode(expectedNode)) {
+			expectedNode.children = expectedChildren;
+		}
+		actualDecoratedNodes.push(actualNode);
+		expectedDecoratedNodes.push(expectedNode);
+	}
+	return [actualDecoratedNodes, expectedDecoratedNodes];
+}
+
 export function assertRender(actual: DNode | DNode[], expected: DNode | DNode[], message?: string): void {
-	const parsedActual = formatDNodes(actual);
-	const parsedExpected = formatDNodes(expected);
+	const [decoratedActual, decoratedExpected] = decorate(actual, expected);
+	const parsedActual = formatDNodes(Array.isArray(actual) ? decoratedActual : decoratedActual[0]);
+	const parsedExpected = formatDNodes(Array.isArray(expected) ? decoratedExpected : decoratedExpected[0]);
```

Comment:
Can you explain what is going on with the `isArray` check? Shouldn't this be changed to the following?
```suggestion
	const parsedExpected = formatDNodes(Array.isArray(decoratedExpected) ? decoratedExpected[0] : decoratedExpected);
```

Reply:
@aciccarello the `decorator` function always turns nodes into an array (if it isn't already) and returns an array. the `assertRender` function and `formatDNodes` take an individual node or an array. so all this does is use the first node if the original nodes (expected and actual) were a singular node. So no the suggested change is incorrect. I might change the decorate function to align with `formatDNodes` to avoid this slightly confusing conversion though.